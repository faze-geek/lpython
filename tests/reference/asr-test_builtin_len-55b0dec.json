{
    "basename": "asr-test_builtin_len-55b0dec",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/test_builtin_len.py",
    "infile_hash": "39ba468d5d0b0ed6fe1bf20a158ba9448e9b799f0277d994588265cd",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-test_builtin_len-55b0dec.stdout",
    "stdout_hash": "b9b214d75e76b8674a48ed168c783ca30fc4f4732003362cc1b20b5c",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}