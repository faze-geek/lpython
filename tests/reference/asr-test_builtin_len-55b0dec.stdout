(TranslationUnit (SymbolTable 1 {_lpython_main_program: (Subroutine (SymbolTable 4 {}) _lpython_main_program [] [(SubroutineCall 1 test_len () [] ())] Source Public Implementation () .false. .false.), main_program: (Program (SymbolTable 3 {}) main_program [] [(SubroutineCall 1 _lpython_main_program () [] ())]), test_len: (Subroutine (SymbolTable 2 {i: (Variable 2 i Local () () Default (Integer 4 []) Source Public Required .false.), s: (Variable 2 s Local () () Default (Character 1 -2 () []) Source Public Required .false.), t: (Variable 2 t Local () () Default (Character 1 -2 () []) Source Public Required .false.)}) test_len [] [(= (Var 2 s) (StringConstant "abcd" (Character 1 4 () [])) ()) (Assert (Compare (StringLen (Var 2 s) (Integer 4 []) ()) Eq (IntegerConstant 4 (Integer 4 [])) (Logical 4 []) () ()) ()) (= (Var 2 s) (StringConstant "" (Character 1 0 () [])) ()) (Assert (Compare (StringLen (Var 2 s) (Integer 4 []) ()) Eq (IntegerConstant 0 (Integer 4 [])) (Logical 4 []) () ()) ()) (Assert (Compare (StringLen (StringConstant "abcd" (Character 1 4 () [])) (Integer 4 []) (IntegerConstant 4 (Integer 4 []))) Eq (IntegerConstant 4 (Integer 4 [])) (Logical 4 []) (LogicalConstant .true. (Logical 4 [])) ()) ()) (Assert (Compare (StringLen (StringConstant "" (Character 1 0 () [])) (Integer 4 []) (IntegerConstant 0 (Integer 4 []))) Eq (IntegerConstant 0 (Integer 4 [])) (Logical 4 []) (LogicalConstant .true. (Logical 4 [])) ()) ()) (= (Var 2 t) (StringConstant "efg" (Character 1 3 () [])) ()) (= (Var 2 s) (StrOp (StringConstant "abcd" (Character 1 4 () [])) Concat (Var 2 t) (Character 1 2 () []) ()) ()) (= (Var 2 i) (StringLen (Var 2 s) (Integer 4 []) ()) ()) (Assert (Compare (Var 2 i) Eq (IntegerConstant 7 (Integer 4 [])) (Logical 4 []) () ()) ()) (= (Var 2 i) (StringLen (StrOp (StringConstant "abc" (Character 1 3 () [])) Concat (StringConstant "def" (Character 1 3 () [])) (Character 1 6 () []) (StringConstant "abcdef" (Character 1 6 () []))) (Integer 4 []) (IntegerConstant 6 (Integer 4 []))) ()) (Assert (Compare (Var 2 i) Eq (IntegerConstant 6 (Integer 4 [])) (Logical 4 []) () ()) ())] Source Public Implementation () .false. .false.)}) [])
